xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/vue/utilities/XWikiIcon.vue

```diff
@@ -39,27 +44,76 @@
 
 
 <script>
+const iconCache = {};
+
 export default {
 
   name: "XWikiIcon",
 
+  props: {
+    iconDescriptor: Object,
+  },
+
+  data () {
+    return {
+      remoteIconDescriptor: undefined,
+    };
+  },
+
   computed: {
     iconSetType () {
-      if (this.iconDescriptor.iconSetType) { return this.iconDescriptor.iconSetType; }
-      else if (this.iconDescriptor.cssClass) { return "FONT"; }
-      else if (this.iconDescriptor.url) { return "IMAGE"; }
+      if (this.iconDescriptor?.iconSetType) { return this.iconDescriptor?.iconSetType; }
+      else if (this.iconDescriptor?.cssClass) { return "FONT"; }
+      else if (this.iconDescriptor?.url) { return "IMAGE"; }
+      else if (this.iconDescriptor?.name) { return "REMOTE"; }
       return "";
     },
   },
 
-  props: {
-    iconDescriptor: Object,
+  methods: {
+    async fetchRemoteIconDescriptor (iconName) {
+      try {
+        const iconURLQuery = `?media=json&name=${iconName}`;
+        const iconURL = `/xwiki/rest/wikis/${XWiki.currentWiki}/iconThemes/icons${iconURLQuery}`;
+        const response = await fetch(iconURL);
+        const jsonResponse = await response.json();
+        iconCache[iconName] = jsonResponse.icons[0];
+        return jsonResponse.icons[0];
+      } catch (err) {
+        console.error(err);
+      }
+    },
```

Comment:
Note: I'll move this method to Logic since this will also be useful for https://jira.xwiki.org/browse/XWIKI-18605. Whoever merges second will need to be careful not to introduce duplicated code.